packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { LibraryCommandletExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src';
+import { notificationService } from '@salesforce/salesforcedx-utils-vscode/out/src/commands';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { format } from 'util';
+import * as vscode from 'vscode';
+import { OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import { FilePathGatherer, SfdxCommandlet, SfdxWorkspaceChecker } from './util';
+
+const RENAME_LIGHTNING_COMPONENT_EXECUTOR = 'force_rename_lightning_component';
+const RENAME_INPUT_PLACEHOLDER = 'rename_component_input_placeholder';
+const RENAME_INPUT_PROMPT = 'rename_component_input_prompt';
+const RENAME_INPUT_DUP_ERROR = 'rename_component_input_dup_error';
+const RENAME_WARNING = 'rename_component_warning';
+
+export class RenameLwcComponentExecutor extends LibraryCommandletExecutor<string> {
+  private sourceFsPath: string;
+  private responseText: string | undefined;
+  constructor(sourceFsPath: string, responseText: string | undefined) {
+    super(
+      nls.localize(RENAME_LIGHTNING_COMPONENT_EXECUTOR),
+      RENAME_LIGHTNING_COMPONENT_EXECUTOR,
+      OUTPUT_CHANNEL
+    );
+    this.sourceFsPath = sourceFsPath;
+    this.responseText = responseText;
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: vscode.Progress<{ message?: string | undefined; increment?: number | undefined; }>,
+    token?: vscode.CancellationToken
+    ): Promise<boolean> {
+      if (this.sourceFsPath) {
+        if (this.responseText) {
+          renameComponent(this.sourceFsPath, this.responseText);
+          return true;
+        }
+      }
+      return false;
+  }
+}
+
+export async function forceRenameLightningComponent(sourceUri: vscode.Uri) {
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(RENAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
```

Comment:
This smells like a service method vs something that should live in the command.  The command code should only really concern itself with command execution.  The check for existing and actual renaming should live elsewhere.  Maybe under a services directory in core or utils.  It looks like we have some one off services here and there but nothing organizing the logic. @jag-sfdc any thoughts or collections of services that I'm unaware of in the codebase?